File: algorithms/python/Challenge_Is_Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        sIdx = 0
        tIdx = 0
        while sIdx < len(s) and tIdx < len(t):
            if t[tIdx] == s[sIdx]:
                sIdx += 1
            tIdx += 1
            
        if sIdx == len(s):
            return True
        
        return False

File: algorithms/java/Challenge_Is_Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sIdx = 0, tIdx = 0;
        int Ns = s.length();
        int Nt = t.length();
        
        while(sIdx < Ns && tIdx < Nt) {
            if(s.charAt(sIdx) == t.charAt(tIdx))
                sIdx++;
            tIdx++;
        }
        
        if(sIdx == Ns)
            return true;
        return false;
    }
}
